xbmc/video/VideoInfoScanner.cpp

```diff
@@ -1990,4 +1995,35 @@ namespace VIDEO
     }
     return 0;    // didn't find anything
   }
+
+  int CVideoInfoScanner::FindVideoUsingTag(CFileItem& pItem, const ScraperPtr &scraper, CScraperUrl &url, CGUIDialogProgress *progress)
+  {
+    MOVIELIST movielist;
+    CVideoInfoDownloader imdb(scraper);
+    // Start searching imdb using title:
+    if ((pItem.IsType(".mkv") || pItem.IsType(".mp4")))
```

Comment:
Use the factory.

Reply:
Right. This will be the biggest argument against this PR I guess. So let's try to summarize the situation here so as to clarify the defined behavior. The factory is currently written such as:

```
  if (CServiceBroker::GetSettings().GetBool(CSettings::SETTING_MYVIDEOS_USETAGS) &&
      (item.IsType(".mkv") || item.IsType(".mp4")))
```
So far so good, the behavior implicitly assume that reading of tags required usage of 

```
CServiceBroker::GetSettings().GetBool(CSettings::SETTING_MYVIDEOS_USETAGS)
```

This seems consistent with the overall user settings. Now the difficult part is that the `CVideoInfoScanner::RetrieveInfoForMovie` is written such that:

```
    // handle .nfo files
    std::unique_ptr<IVideoInfoTagLoader> loader;
    if (useLocal)
    {
      loader.reset(CVideoInfoTagLoaderFactory::CreateLoader(*pItem, info2, bDirNames));
      if (loader)
      {
        pItem->GetVideoInfoTag()->Reset();
        result = loader->Load(*pItem->GetVideoInfoTag(), false);
      }
    }
    if (result == CInfoScanner::FULL_NFO)
    {
      if (AddVideo(pItem, info2->Content(), bDirNames, true) < 0)
        return INFO_ERROR;
      return INFO_ADDED;
    }
```

What this means, is that as soon as a user setup the Video Library with boolean `SETTING_MYVIDEOS_USETAGS` the function FindVideo is never called, hence I can never query imdb to retrieve the video summary of thumbnail to display in the UI anymore.

Let me know how best to cope with this design, I guess either a refactor of the factory, or another change in the `RetrieveInfoForMovie` behavior. I am new to kodi, so I tried not to break old behavior.

Reply:
@notspiff I've seen your other suggestions and will include them, but fur now I'll hold any further change until this main question is solved. Thanks

Reply:
ah right, that makes a lot of sense. or well, the reason you had to do this makes sense.

it's tricky to know how to do this properly, as it's hard to know whether a user thinks 'title only' is good enough. do you have other tags in your use case or would 'if title only, return a new entry in CInfoScanner::INFO_TYPE 'TITLE_NFO' or the likes work ?

Reply:
<deep breath>From my understanding and looking at some other posts online, we should be able to get near a perfect match if we query based on 'title' & 'year'. For example:
* https://answers.microsoft.com/en-us/windows/forum/windows_7-pictures/editing-tagsmeta-datafile-properties-on-large-mp4/8e502eed-33fe-4585-9ca2-23fb9114481d
> I have found that just editing the "TITLE" and "YEAR" attributes/tags/metadata on the .MP4 files allow WMC7 to natively pull down descriptions and cover art.
While my current patch only implement 50% of my long term goal, my goal is to query imdb using both title & year. The goal is to adressed the issue described in the wiki:
* https://kodi.wiki/view/NFO_files#Parsing_nfo
> As an example, searching for The Mummy on TMDB returns 65 titles.
Using both title and year should allow a near perfect match.

Reply:
okay, so we can do what i suggest but allow year and title (but nothing else). right? that is basically the same stuff that is output from the regular expression for the file name (a lot of hoops takes us to CUtil::CleanString). some api changes will be needed, ie, we'd have to lift the year parsing a bit further up the stack and then pass it down through the FindVideo call.

Reply:
OK, make sense. I'll prepare a new set of patches based on my understanding (changing FindVideo API). I am not clear about your suggestion for ` CInfoScanner::INFO_TYPE 'TITLE_NFO`. Anyway this may become clear as I prepare the changes.

Reply:
you would use that to go through the factory, yet still enable a call to FindVideo.

Reply:
I believe the new patchset should match your suggested implementation. Let me know if this is the right directlion and I'll do the final cleanup pass (indentation, sync musicvideo...)